### PR TITLE
Add Float16 Stream Data Type

### PR DESCRIPTION
**Pull request by myungjoo**
TODO
- [x] Unit tests
- [x] Test Result / aarch64
- [x] Test Result / arm
- [x] Test Result / x64
- [x] Test Result / x86
- [x] Check availability in x86_32/x64 ==> NEED GCC12. Do it later.
- [x] Negative cases

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
The DOXYGEN error is false positive!

**Comment by myungjoo** [comment]
Status: added unit tests. Failing...

File: packaging/nnstreamer.spec [hunk 1]
```diff
@@ -871,10 +885,14 @@ export NNSTREAMER_CONVERTERS=${NNSTREAMER_BUILD_ROOT_PATH}/ext/nnstreamer/tensor
...
 
+%if 0%{?fp16_enabled}
+    ## If fp16 tests fail, stop the build!
+    export FLOAT16_SUPPORTED=1
+%endif
     ssat -n -p=1 --summary summary.txt -cn _n
     popd
 
-python3 tools/development/count_test_cases.py build tests/summary.txt
+#python3 tools/development/count_test_cases.py build tests/summary.txt
```
Comment (jaeyun-jung) [inline packaging/nnstreamer.spec:895]: Q: why did you blocked this line?

File: ext/nnstreamer/tensor_filter/vivante/tensor_filter_subplugin.c [hunk 2]
```diff
@@ -245,11 +245,15 @@ convert_tensortype (unsigned tensor_type)
...
     case VSI_NN_TYPE_UINT16:
       return _NNS_UINT16;
-  /** Note that the current nnstreamer (tensor_typedef.h) does not support FLOAT16.
-   *  Let's use UINT16 as a workaround.
-   */
     case VSI_NN_TYPE_FLOAT16:
+#ifdef FLOAT16_SUPPORT
+      return _NNS_FLOAT16;
+#else
+  /**
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/vivante/tensor_filter_subplugin.c:252]: need fix indent

**Review by anyj0527 — APPROVED** [review]
LGTM otherwise jaeyun's comment

File: packaging/nnstreamer.spec (hunk 1 above)
Comment (myungjoo) [reply packaging/nnstreamer.spec:895]: it was mistake. thanks.

File: ext/nnstreamer/tensor_filter/vivante/tensor_filter_subplugin.c (hunk 2 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/vivante/tensor_filter_subplugin.c]: fixed.

**Review by wooksong — APPROVED** [review]
LGTM!

**Pull request merged**